tests/test_packages/functional/package_test/__init__.py

```diff
@@ -0,0 +1,273 @@
+    if os.path.isfile(suppress_file):
+        os.remove(suppress_file)
+    _generate_suppress_file(suppress_file)
+
+    # generate skip list file
+    skip_list_file = os.path.join(tmp_dir, 'skip_list_file')
+    if os.path.isfile(skip_list_file):
+        os.remove(skip_list_file)
+    _generate_skip_list_file(skip_list_file)
+
+    shared_test_params = {
+        'suppress_file': suppress_file,
+        'env': env,
+        'use_postgresql': use_postgresql,
+        'workspace': workspace,
+        'database': database
+    }
+
+    # first check
+    _clean_project(test_project_path, test_project_clean_cmd, env)
+    test_project_1_name = project_info['name'] + '_' + uuid.uuid4().hex
+
+    _run_check(shared_test_params, skip_list_file, test_project_build_cmd,
+               test_project_1_name, test_project_path)
+
+    time.sleep(5)
+
+    # second check
+    _clean_project(test_project_path, test_project_clean_cmd, env)
+
+    test_project_2_name = project_info['name'] + '_' + uuid.uuid4().hex
+
+    _run_check(shared_test_params, skip_list_file, test_project_build_cmd,
+               test_project_2_name, test_project_path)
+
+    time.sleep(5)
+
+    # start the CodeChecker server
+    _start_server(shared_test_params, test_config)
+
+
+def teardown_package():
+    """Stop the CodeChecker server."""
+    __STOP_SERVER.set()
+
+    time.sleep(10)
+
+
+def _clean_project(test_project_path, clean_cmd, env):
+    """Clean the test project."""
+    command = ['bash', '-c', clean_cmd]
+
+    try:
+        subprocess.check_call(command, cwd=test_project_path, env=env)
+    except subprocess.CalledProcessError as perr:
+        raise perr
+
+
+def _generate_suppress_file(suppress_file):
+    """
+    Create a dummy supppress file just to check if the old and the new
+    suppress format can be processed.
+    """
+    import calendar
+    import hashlib
+    import random
+
+    hash_version = '1'
+    suppress_stuff = []
+    for _ in range(10):
+        curr_time = calendar.timegm(time.gmtime())
+        random_integer = random.randint(1, 9999999)
+        suppress_line = str(curr_time) + str(random_integer)
+        suppress_stuff.append(
+            hashlib.md5(suppress_line).hexdigest() + '#' + hash_version)
+
+    s_file = open(suppress_file, 'w')
+    for k in suppress_stuff:
+        s_file.write(k + '||' + 'idziei éléáálk ~!@#$#%^&*() \n')
+        s_file.write(
+            k + '||' + 'test_~!@#$%^&*.cpp' +
+            '||' 'idziei éléáálk ~!@#$%^&*(\n')
+        s_file.write(
+            hashlib.md5(suppress_line).hexdigest() + '||' +
+            'test_~!@#$%^&*.cpp' + '||' 'idziei éléáálk ~!@#$%^&*(\n')
+
+    s_file.close()
+
+
+def _generate_skip_list_file(skip_list_file):
+    """
+    Create a dummy skip list file just to check if it can be loaded.
+    Skip files without any results from checking.
+    """
+    skip_list_content = []
+    skip_list_content.append("-*randtable.c")

```

Comment:
This examples can be changed. Old bzip2 file names are in the examples.
A new test source file could be added (with a bug in it) and skipped which will test the skip functionality.